packages/eslint-plugin-orbit-components/src/rules/themeRTL.ts

```diff
@@ -0,0 +1,63 @@
+import * as t from "@babel/types";
+import type { Rule } from "eslint";
+
+export const ERROR =
+  "Do not use theme.rtl conditions. Import RTL utility functions from Orbit and use them instead";
+
+export default {
+  meta: {
+    type: "problem",
+    docs: {
+      description: "Prevents bad RTL patterns",
+      category: "Possible Errors",
+      recommended: true,
+    },
+  },
+
+  create: (context: Rule.RuleContext) => {
+    let specifier = "";
+
+    return {
+      ImportDeclaration(node: t.ImportDeclaration) {
+        if (node.source.value === "styled-components") {
+          const def = node.specifiers.filter(s => t.isImportDefaultSpecifier(s));
+          specifier = def[0].local.name;
+        }
+      },
+
+      TaggedTemplateExpression(node: t.TaggedTemplateExpression) {
+        if (t.isMemberExpression(node.tag)) {
+          if (t.isIdentifier(node.tag.object) && node.tag.object.name === specifier) {
+            if (t.isTemplateLiteral(node.quasi)) {
+              node.quasi.expressions.forEach(e => {
+                if (t.isArrowFunctionExpression(e)) {
+                  if (t.isConditionalExpression(e.body)) {
```

Comment:
We should be more specific when comes to the conditional expression, otherwise we would prohibit the usage of arrow functions that returns conditional expression completely.

The rule will also report this case:

```jsx
const StyledDiv = styled.div`
  ${({ someCustomProperty }) => someCustomProperty ? "right" : "left"};
`
```

I would try to implement it only for conditional expression that uses `theme.rtl` or `rtl` (if it's destructurized) as the  `test` property of `ConditionalExpression`.

Also please add some generic use case that uses conditional expressions to the valid set of tests 🙏